### PR TITLE
fix: prevent crash if BrowserView webContents was destroyed

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Fixes an issue where interacting with a BrowserWindow which had a BrowserView attached whose `webContents` had been destroyed could cause a crash in `BaseWindow::ResetBrowserViews()`.

cc @nornagon @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).
- [x] This is **NOT A BREAKING CHANGE**. Breaking changes may not be merged to master until 11-x-y is branched.

#### Release Notes

Notes: none.

### CODE REVIEW

**Comment by codebytere** [comment]
This might be more complex than originally anticipated bc `BrowserView::WebContentsDestroyed()` is destroyed async as of https://github.com/electron/electron/commit/746beb0d8b28809ec96d6ce9ad1ee252402584a7 so when we check for `browser_view->web_contents()` not being nullptr it might falsely still show as existing so this fix isn't deterministic

File: docs/api/browser-view.md [hunk 1]
```diff
@@ -40,16 +40,6 @@ A [`WebContents`](web-contents.md) object owned by this view.
 
 Objects created with `new BrowserView` have the following instance methods:
 
-#### `view.destroy()`
```
Comment (codebytere) [inline docs/api/browser-view.md:43]: this does not exist

**Review by zcbenz — APPROVED** [review]

**Comment by vladimiry** [comment]
@codebytere 

> This is NOT A BREAKING CHANGE. Breaking changes may not be merged to master until 11-x-y is branched.

The change looks like a breaking change for me since some public methods got removed.

How do I explicitly destroy the BrowserView instance in v11? Does the linked BrowserView isntance get automatically destroyed when I call `browserWindow.setBrowserView(null)`?

**Comment by codebytere** [comment]
@vladimiry this PR didn't remove the methods - it followed up on another PR which did so. However, we do agree with you - the PR which did remove these methods did not align with above stated goals as you point out and the methods will be restored.

**Comment by vladimiry** [comment]
> this PR didn't remove the methods - it just addressed documentation as a follow-up on another PR which did so

Thanks for clarification, I missed looking into that another PR.

**Pull request merged**